packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -360,28 +360,33 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded> {
         label: Text(appLocalizations.nutrition_page_add_nutrient),
       );
 
-  Future<bool> _showCancelPopup(AppLocalizations localizations) async =>
-      await showDialog<bool>(
-        context: context,
-        builder: (BuildContext context) => AlertDialog(
-          shape: const RoundedRectangleBorder(
-            borderRadius: ROUNDED_BORDER_RADIUS,
-          ),
-          title: Text(localizations.general_confirmation),
-          content: Text(localizations.nutrition_page_close_confirmation),
-          actions: <TextButton>[
-            TextButton(
-              child: Text(localizations.cancel.toUpperCase()),
-              onPressed: () => Navigator.pop(context, false),
-            ),
-            TextButton(
-              child: Text(localizations.close.toUpperCase()),
-              onPressed: () => Navigator.pop(context, true),
+  Future<bool> _showCancelPopup(AppLocalizations localizations) async {
```

Comment:
If you have 2 minutes, just add a comment to explain the returned values.
It's not intuitive to understand what a "show cancel popup" is supposed to return, especially if not only you return a bool but you pop a bool too.

Reply:
Corrected the code, plz  review the code again, 
made some changes, just returning a bool true for the WillPopScope  does the job

Reply:
But you're still returning 2 different bools. If one of them is useless, please don't use it. And please comment in the code, not here, something like a one-line `///`.

Reply:
I am not sure why it is still that confusing , have written comments, please have a look again